test/unix/main.ml

```diff
@@ -143,23 +238,86 @@ let readonly () =
               (Printf.sprintf "Wrong insertion: %s value is missing." v))
     !l
 
+let readonly_no_flush () =
+  let w =
+    Index.v ~fresh:true ~readonly:false ~log_size ~fan_out_size index_name
+  in
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  List.iter (fun (k, v) -> Index.add w k v) !l;
+  test_find_present r
+
+let readonly_lwt () =
+  let w =
+    Index.v ~fresh:true ~readonly:false ~log_size ~fan_out_size index_name
+  in
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  let wp = Lwt_list.iter_p (fun (k, v) -> Lwt.return (Index.add w k v)) !l in
+  let p =
+    Lwt.bind
+      (Lwt.bind wp (fun _ -> Lwt.return (Index.flush w)))
+      (fun _ -> test_find_present_lwt r)
+  in
+  Lwt_main.run p
+
+let fail_readonly_add () =
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  try
+    let () = Index.add r (Key.v ()) (Value.v ()) in
+    Alcotest.fail (Printf.sprintf "Index readonly cannot write.")
+  with _ -> ()
+
+let test_clear () =
+  let () = Index.clear t in
```

Comment:
I'm uncomfortable with this test depending on a global `t`. It suggests that the tests aren't robust to being selectively executed, but perhaps you don't want to initialise a new index for some reason?

Reply:
Actually I believe the tests depend on each other, so they can't be run separately, witch was a pretty bad design from me. It probably needs a separate PR to handle this.

Reply:
Not a big deal, but my experiences from debugging Irmin tests are that it's nice to leverage individual test execution where possible. Perhaps this is a situation that would benefit from an explicit 'test with context' feature in Alcotest.